cmd/zc_enumerator.go

```diff
@@ -432,8 +432,8 @@ func InitResourceTraverser(resource common.ResourceString, location common.Locat
 
 		recommendHttpsIfNecessary(*resourceURL)
 
-		if ctx == nil || p == nil {
-			return nil, errors.New("a valid credential and context must be supplied to create a blob traverser")
+		if ctx == nil {
```

Comment:
All the case statements, including at line 416, require a context. This check can be advanced to the start of function.

Reply:
hmm it looks like benchmark and local don't require a context

Reply:
We can consider this after migrating all 3 services. This isn't too big of a deal in terms of changes